### PR TITLE
DRMAAV2-27: Infrastructure for drmaa2_dict

### PR DESCRIPTION
**Pull request by vinodchitrali**
#### Issue-ID

* **[DRMAAV2-27](https://pbspro.atlassian.net/browse/DRMAAV2-27)**

#### Problem description
* Key value pair dictionary  drmaa2

#### Cause / Analysis
* None

#### Solution description
* none

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **unit test(s) to my commit**. *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why unit testing is not appropriate** for this case.
- [x] All new and existing automated tests have passed. 
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: api/c-binding/drmaa2.cpp [hunk 1]
```diff
@@ -392,8 +444,16 @@ drmaa2_string_list drmaa2_dict_list(const drmaa2_dict d) {
  *
  */
 drmaa2_bool drmaa2_dict_has(const drmaa2_dict d, const char *key) {
-	//TODO Add Code here
-	return DRMAA2_TRUE;
+	if (d == NULL)
```
Comment (rampranesh) [inline api/c-binding/drmaa2.cpp]: We need to check NULL for key as well, same in all applicable cases

Comment (vinodchitrali) [inline api/c-binding/drmaa2.cpp]: fixed

**Comment by vinodchitrali** [comment]
maintainers, Please merge this PR

File: test/unittesting/src/Makefile.am [hunk 2]
```diff
@@ -41,9 +41,10 @@ test_drmaa_SOURCES=	SessionManagerTest.cpp \
 			PBSProSystem_test.cpp \
 			drmaa2listTest.cpp \
 			JobSessionTest.cpp \
+			drmaa2dictTest.cpp \
 			runtest.cpp
 						
 test_drmaa_LDADD = ../../../api/libdrmaav2.la -lcppunit
 
-test_drmaa_CPPFLAGS = -I ../inc -I$(top_srcdir)/api/c-binding -I$(top_srcdir)/api/cpp-binding -I$(top_srcdir)/inc -I$(drms_inc_dir)
+test_drmaa_CPPFLAGS = -fPIC -I ../inc -I$(top_srcdir)/api/c-binding -I$(top_srcdir)/api/cpp-binding -I$(top_srcdir)/inc -I$(drms_inc_dir)
```
Comment (mike0042) [inline test/unittesting/src/Makefile.am]: Is there a specific reason for forcing position independent code? In most cases you're better off letting libtool take care of these parameters for you. Shared libraries will be compiled as position independent code under Linux without having to specify -fPIC.

Comment (vinodchitrali) [inline test/unittesting/src/Makefile.am]: fixed

**Comment by vinodchitrali** [comment]
@mike0042 . Please review

**Pull request merged**